src/app/locator/qgsgotolocatorfilter.cpp

```diff
@@ -87,51 +99,109 @@ void QgsGotoLocatorFilter::fetchResults( const QString &string, const QgsLocator
     }
   }
 
-  if ( okX && okY )
+  if ( okX && okY )  // Ok found a valid looking x,y coordinate pair, check against Wgs84crs, MapCanvas-crs AND layer-crs's
   {
-    QVariantMap data;
-    const QgsPointXY point( posX, posY );
-    data.insert( QStringLiteral( "point" ), point );
-
-    const bool withinWgs84 = wgs84Crs.bounds().contains( point );
-    if ( !posIsDms && currentCrs != wgs84Crs )
+    // collect all crs's from all layers (plus MapCanvas crs and wgs84)
+    QList<QgsCoordinateReferenceSystem> crsList;
+    crsList.append( wgs84Crs );
+    if ( currentCrs != wgs84Crs )
+      crsList.append( currentCrs );
+    // append optional extra crs (added via config)
+    if ( extraCrs != wgs84Crs )
```

Comment:
```suggestion
    if ( !crsList.contains( extraCrs ) )
```
to avoid duplicated entries